### PR TITLE
Remove Managed Ledger metadata text format

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Complete the steps described in #281.

In v1.17, (#282) ManagedLedger started supporting both text and binary versions of protobuf encoding for the metadata in ZK.

In v1.18, we can leave only the option for binary format. If a rollback to 1.17 becomes necessary, it won't be a problem since 1.17 can deserialize the binary format.

### Modifications

Make binary format the only option for managed ledger metadata.

### CODE REVIEW

**Comment by merlimat** [comment]
@rdhabalia @saandrews Please take a look

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java [hunk 1]
```diff
@@ -330,60 +316,29 @@ public void removeManagedLedger(String ledgerName, MetaStoreCallback<Void> callb
...
-            } catch (InvalidProtocolBufferException e) {
-                return parseManagedLedgerInfoFromText(data);
-            }
+        // First try binary format, then fallback to text
+        try {
+            return ManagedLedgerInfo.parseFrom(data);
+        } catch (InvalidProtocolBufferException e) {
+            // Fallback to parsing protobuf text format
+            ManagedLedgerInfo.Builder builder = ManagedLedgerInfo.newBuilder();
+            TextFormat.merge(new String(data, Encoding), builder);
```
Comment (saandrews) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java]: We should handle it even if parsing text fails.

**Comment by saandrews** [comment]
The change looks good. We have to hold this pull request until we are ready with all the dependent tools to handle binary format.

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java (hunk 1 above)
Comment (merlimat) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java]: What do you mean? If we cannot parse the protobuf we need to throw exception..

**Comment by merlimat** [comment]
> The change looks good. We have to hold this pull request until we are ready with all the dependent tools to handle binary format.

The tool to get the ML metadata in text format is already in 1.17: 
```
bin/pulsar-admin persistent info-internal $MY_TOPIC
```

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java (hunk 1 above)
Comment (saandrews) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImplZookeeper.java]: I didn't notice that the method is throwing exception. I was thinking that we are not handling the exception when it fails here.

**Comment by saandrews** [comment]
Well, we have some internal tools as well :) I will merge it as soon as we are ready.

**Comment by merlimat** [comment]
@saandrews any update on this one? Should we move to 1.19?

**Comment by saandrews** [comment]
@merlimat Yeah, lets move it to 1.19

**Comment by merlimat** [comment]
@saandrews sure, but let's try to get this in soon. This is reducing the ZK metadata footprint considerably. Also this gives freedom to update the metadata format without worrying about backward compatibility issues.

**Comment by merlimat** [comment]
@saandrews @rdhabalia Since #526 is already in, can we merge this one?

**Comment by merlimat** [comment]
@saandrews @rdhabalia Ping

**Comment by rdhabalia** [comment]
retest this please

**Comment by merlimat** [comment]
@saandrews @rdhabalia Any update on this? 

I'd like to add some new optional fields in the cursor protobuf data and it would be difficult to do it with the text format (for the schema migration and deployment rollback).

**Comment by saandrews** [comment]
Maybe we can merge and defer this in our branch until we are ready(though won't prefer branching).  @joefk  ?

**Comment by joefk** [comment]
I am good with merging this.

**Comment by merlimat** [comment]
> I am good with merging this.

@joefk Still need "formal" PR approval, otherwise it cannot be merged 😃

**Comment by merlimat** [comment]
🎉

**Pull request merged**